File: w03/d05/Nicholas/cipher/lib/cipher.rb
class Cipher

  def self.encode(word)
    new_word = word.gsub(/[abcdefghijklmnopqrstuvwxyz]/, 'a'=> 'n', 'b' => 'o', 'c' => 'p', 'd' => 'q', 'e' => 'r', 'f' => 's', 'g' => 't', 'h' => 'u', 'i' => 'v', 'j' => 'w', 'k' => 'x', 'l' => 'y', 'm' => 'z', 'n' => 'a', 'o' => 'b', 'p' => 'c', 'q' => 'd', 'r' => 'e', 's' => 'f', 't' => 'g', 'u' => 'h', 'v' => 'i', 'w' => 'j', 'x' => 'k', 'y' => 'l', 'z' => 'm')
    return new_word
  end
  
end

File: w08/d03/Tahira/vignere_cipher/cipher.js

// var message = prompt("Please enter the message you'd like to encode:");
// var keyword = prompt("Enter the keyword:");

var Cipher = function(message, keyword) {
  this.message = message.toLowerCase();
  this.keyword = keyword.toLowerCase();
};

Cipher.prototype.encode = function() {
  alpha = "abcdefghijklmnopqrstuvwxyz".split("");
  messageArray = this.message.split("");
  keywordArray = this.keyword.split("");
  var encryptedMessage = "";
  var k = 0;
  var j = 0;
  var repeatKey = "";

  while(j < this.message.length) {
    repeatKey += keywordArray[k];
    k++;  
    if (k > this.keyword.length-1){
      k = 0;
    };
    j++;
  };

  for(var i=0; i<this.message.length; i++) {
    messageNum = alpha.indexOf(messageArray[i]);
    keyNum = alpha.indexOf(repeatKey[i]);
    sum = messageNum + keyNum;
    if (sum > 25) {
      sum = sum - 26;
    }; 
    encryptedMessage += alpha[sum];
  };
  return encryptedMessage;
};

Cipher.prototype.decode = function() {
  console.log("Hello");
};

cipher1 = new Cipher("attackatdawn", "lemon");
console.log(cipher1.encode());

Cipher.prototype.decode = function() {
  alpha = "abcdefghijklmnopqrstuvwxyz".split("");
  messageArray = this.message.split("");
  keywordArray = this.keyword.split("");
  var encryptedMessage = "";
  var k = 0;
  var j = 0;
  var repeatKey = "";

  while(j < this.message.length) {
    repeatKey += keywordArray[k];
    k++;  
    if (k > this.keyword.length-1){
      k = 0;
    };
    j++;
  };

  for(var i=0; i<this.message.length; i++) {
    messageNum = alpha.indexOf(messageArray[i]);
    keyNum = alpha.indexOf(repeatKey[i]);
    sum = messageNum - keyNum;
    if (sum < 0) {
      sum = sum + 26;
    }; 
    encryptedMessage += alpha[sum];
  };
  return encryptedMessage;
};

cipher2 = new Cipher(cipher1.encode(), "lemon");
console.log(cipher2.decode());